index.html

```diff
@@ -5007,13 +4983,259 @@ <h3 id="looping-AudioBufferSourceNode">
+                audio block to be rendered:
+                  <ol>
+                    <li>If <code>currentTime &lt; start</code> or
+                    <code>currentTime &gt;= stop</code>, emit silence for the
+                    output frame at <code>index</code>.
+                    </li>
+                    <li>Else,
+                      <ol>
+                        <li>If <code>bufferTime</code> &lt; 0, set
+                        <code>bufferTime</code> to <code>offset + ((currentTime
+                        - start) * computedPlaybackRate)</code>. This is the
+                        <a>playhead position</a> for the first output sample
+                        frame.
+                        </li>
+                        <li>Let <code>effectiveBufferTime</code> equal
+                        <code>bufferTime</code>.
+                        </li>
+                        <li>map the playhead position to the correct point
+                        within the loop body accounting for loop iterations:
+                          <ol>
+                            <li>While <code>loop</code> is true and
+                            <code>effectiveBufferTime</code> &gt;=
+                            <code>actualLoopEnd</code>,
+                              <ol>
+                                <li>Let <code>effectiveBufferTime</code> equal
+                                <code>effectiveBufferTime - (actualLoopEnd -
+                                actualLoopStart)</code>.
+                                </li>
+                              </ol>
+                            </li>
+                          </ol>
+                        </li>
+                        <li>Emit the result of
+                        <code>playbackSignal(effectiveBufferTime)</code> as the
+                        output frame at <code>index</code>.
+                        </li>
+                        <li>Increase <code>bufferTime</code> by <code>dt *
+                        computedPlaybackRate</code>.
+                        </li>
+                      </ol>
+                    </li>
+                    <li>Increase <code>index</code> by 1.
+                    </li>
+                    <li>Increase <code>currentTime</code> by <code>dt</code>.
+                    </li>
+                  </ol>
+                </li>
+                <li>If <code>currentTime</code> &gt;= <code>stop</code>,
+                consider playback for the node to have ended.
+                </li>
+              </ol>
+            </li>
+          </ol>
+          <p>
+            The following figures illustrate the behavior of the algorithm in
+            assorted key scenarios. Buffer optimization is not considered, but
+            by definition it does not materially affect the resulting playback.
+            In all figures, the following conventions apply:
+          </p>
+          <ul>
+            <li>context sample rate is 1000 Hz
```

Comment:
I think the math and diagrams would be slightly easier if the context rate was 1 Hz and not 1000 Hz.  Or maybe specify the times and stuff in the diagrams using ms units instead of sec?